### ISSUE TITLE
feat: support storybook composition

### ISSUE THREAD

**Issue by RobbyRabbitman** [Sidnioulz/storybook-addon-tag-badges#17]
Hi, thanks for the great addon!

I wonder if displaying the badges in a storybook composition setup is possible - probably not because the host just queries the metadata of the components and stories of the other storybook instances right?

**Comment by Sidnioulz** [comment]
Hi @RobbyRabbitman!

For sure, tagBadges addons defined on the composed Storybooks would not load (https://storybook.js.org/docs/sharing/storybook-composition says somewhere that composed addons do not get loaded).

I don't know if addons defined on the Storybook instance that loads external stories run everywhere. I also don't know if the sidebar entries go through the same code, and that's something I'd need to try out.

If you have a public repo set up with composition already configured, feel free to share it. It'll save me some time to repro and test.

**Comment by RobbyRabbitman** [comment]
Thanks for the reply, I will try it out in the next couple of days but i cannot share it 😄

**Comment by Sidnioulz** [comment]
Thanks! I'm assigning this to you so I know I'm waiting to hear back from you :)

**Comment by RobbyRabbitman** [comment]
Observation: the badges are shown when the config is also set on the host storybook instance - however I wonder why the output is different. Left=host and Right=Child. Both instances use the framework `@storybook/web-components-vite` and set exactly the same config, because its in a shared package. I haven't got the time to debug it yet.

![](https://github.com/user-attachments/assets/2e86a8df-d99b-4649-8019-38eb2cb39d69)

**Comment by Sidnioulz** [comment]
Could you please check lock files for both projects to check they use the
same version? Just to eliminate the matter of a recent update I did that
changes display conditions for stories badges.

**Comment by Sidnioulz** [comment]
Looking back on this, I believe that when you host Storybook instances, it's the config of the host instance that is applied to addons throughout. This might explain why you have a different outcome when composing in a host vs directly loading those stories.

**Comment by marksy** [comment]
@RobbyRabbitman did you get this working? If so, how please?

**Comment by RobbyRabbitman** [comment]
No, it does not work, but I did not spend time debugging it. We will soon go to sb9 - maybe it just works with the new major

**Comment by Sidnioulz** [comment]
@RobbyRabbitman if you can share a repro repository, I could check if sidebar items in composed stories vs the host instance:
* have the same item types
* have tags loaded properly
* have parent - child relationships loaded properly

I think the explanation lies in how Storybook composes portable stories. I do not believe Storybook 9 should be any different from 8.6 in that regard.

**Comment by aweber1** [comment]
I encountered this behavior as well and put together a patch in #168 

This allows adding the `addons: ['storybook-addon-tag-badges']` to the composition host `main.ts` file, e.g.
```
const config: StorybookConfig = {
  stories: ['./*.stories.@(ts|tsx)'],
  addons: ['storybook-addon-tag-badges'],
  framework: '@storybook/react-vite',
  refs: {
    storybook1: {
      title: 'Storybook 1',
      url: '[your url here]',
      expanded: false,
    },
    storybook2: {
      title: 'Storybook 2',
      url: '[your url here]',
      expanded: false,
    },
  },
}
```

And then when browsing to stories from the composition host, badges are rendered at the component level and not for each story.

Note: as expected, you still need the `'storybook-addon-tag-badges'` addon defined in `main.ts` for the individual storybook instances.

**Comment by Sidnioulz** [comment]
Thanks for the PR @aweber1! @RobbyRabbitman please let me know if this fixes the issue for you.
